src/client/scripts/game/chess/specialmove.js

```diff
@@ -9,116 +9,121 @@ import coordutil from '../misc/coordutil.js';
-        }
-
-        // Special move was executed!
-        // There is no captured piece with castling
-        return true;
-    },
-
-    pawns(gamefile, piece, move, { updateData = true, animate = true, updateProperties = true, simulated = false } = {}) {
-
-        // If it was a double push, then add the enpassant flag to the gamefile, and remove its special right!
-        if (updateProperties && specialmove.isPawnMoveADoublePush(piece.coords, move.endCoords)) {
-            gamefile.enpassant = specialmove.getEnPassantSquare(piece.coords, move.endCoords);
-        }
-
-        const enpassantTag = move.enpassant; // -1/1
-        const promotionTag = move.promotion; // promote type
-        if (!enpassantTag && !promotionTag) return false; ; // No special move to execute, return false to signify we didn't move the piece.
-
-        const captureCoords = enpassantTag ? specialmove.getEnpassantCaptureCoords(move.endCoords, enpassantTag) : move.endCoords;
-        const capturedPiece = gamefileutility.getPieceAtCoords(gamefile, captureCoords);
-
-        if (capturedPiece) move.captured = capturedPiece.type;
-        if (capturedPiece && simulated) move.rewindInfo.capturedIndex = capturedPiece.index;
-
-        // Delete the piece captured
-        if (capturedPiece) movepiece.deletePiece(gamefile, capturedPiece, { updateData });
-
-        if (promotionTag) {
-            // Delete original pawn
-            movepiece.deletePiece(gamefile, piece, { updateData });
-
-            movepiece.addPiece(gamefile, promotionTag, move.endCoords, null, { updateData });
-
-        } else /* enpassantTag */ {
-            // Move the pawn
-            movepiece.movePiece(gamefile, piece, move.endCoords, { updateData });
-        }
-
-        if (animate) animation.animatePiece(piece.type, piece.coords, move.endCoords, capturedPiece);
-
-        // Special move was executed!
-        return true;
-    },
-
-    isPawnMoveADoublePush(pawnCoords, endCoords) { return Math.abs(pawnCoords[1] - endCoords[1]) === 2; },
-
-    /**
-     * Returns the en passant square of a pawn double push move
-     * @param {number[]} moveStartCoords - The start coordinates of the move
-     * @param {number[]} moveEndCoords - The end coordinates of the move
-     * @returns {number[]} The coordinates en passant is allowed
-     */
-    getEnPassantSquare(moveStartCoords, moveEndCoords) {
-        const y = (moveStartCoords[1] + moveEndCoords[1]) / 2;
-        return [moveStartCoords[0], y];
-    },
-
-    // MUST require there be an enpassant tag!
-    getEnpassantCaptureCoords(endCoords, enpassantTag) { return [endCoords[0], endCoords[1] + enpassantTag]; },
-};
-
-export default specialmove;
\ No newline at end of file
+export default {
+    getFunctions,
+    kings,
+    pawns,
+    isPawnMoveADoublePush,
+    getEnPassantSquare,
+    getEnpassantCaptureCoords,
```

Comment:
getFunctions() is the only method called externally, all others don't need to be exported, but kept private.